Session1 challenge - Elbert Castañeda

elbertcastaneda opened the pull request:
all in all !!!

Commit:
session1-challenge1 succesfully by Elbert Castañeda

Commit:
Update the state.date only if the data.temperature is different, i forgot this TODO :)

charliesbox commented on session1/challenge/client/src/js/components/container/App.js:
```diff
@@ -27,28 +31,34 @@ const DataTile = ({ name, value, loading, active, onClick }) => {
    * active={key === principal}*/
-
-  return null
+  return Object.keys(items).map((key) =>{
```
another way to do it is by using `Object.entries`, so you can avoid using `items[key]` to get the data

``` js
Object.entries(items).map(([key, item]) => (
     <TileErrorBoundary key={key}>
        <DataTile 
          value={item}
          name={key}
          onClick={onClick}
          loading={loading}
          active={key === principal} />
      </TileErrorBoundary>
    )
);
```

elbertcastaneda replied:
Ok, i got it !!!.

I was always work with **"keys"** method, but **"entries"** is really better, because we get the each element directly, that really is very good.

Thank you so much for your comments

charliesbox commented on session1/challenge/client/src/js/components/container/App.js:
```diff
@@ -133,12 +143,21 @@ class App extends Component {
 
-    socket.on("FromAPI", (data) => {
-      this.setState(() => null);
+      this.setState(() =>  {
```
``` js
this.setState(() => ({ data });
```

charliesbox commented on session1/challenge/client/src/js/components/presentational/TileErrorBoundary.js:
```diff
@@ -1,6 +1,22 @@
 import { DataTileStyled } from '../container/App.styled';
 
-const TileErrorBoundary = null;
+class TileErrorBoundary extends React.Component {
```
you already have `Component` here, so you can avoid using `React.Component` :)

elbertcastaneda replied:
Ok, i forgot to use **Component**, instead of **React.Component**, it was an oversight, but thank you for your comment.

elbertcastaneda committed:
I removed the React object from the class to extend

elbertcastaneda committed:
I implemented the "entries" method instead of "keys"

Robfz commented on session1/challenge/client/src/js/components/presentational/TileErrorBoundary.js:
```diff
@@ -1,6 +1,22 @@
+    this.state = { hasError: false };
+  }
+
+  componentDidCatch(error, info) {  
```
Since you are not using the `info` parameter, you can delete it.

Robfz commented on session1/challenge/client/src/js/components/container/App.js:
```diff
@@ -27,28 +31,34 @@ const DataTile = ({ name, value, loading, active, onClick }) => {
   );
 
-  return null;
+  return ReactDOM.createPortal(modal, document.getElementsByTagName('body')[0]);;
```
There is a file named `ModalPortal.js` inside the `presentational` folder. Try to implement the `Portal` component there.

elbertcastaneda replied:
Ok, working on that

Robfz commented on session1/challenge/client/src/js/components/container/App.js:
```diff
@@ -133,12 +143,21 @@ class App extends Component {
 
-    socket.on("FromAPI", (data) => {
-      this.setState(() => null);
+      this.setState(() =>  {
```
Why do you have another call to `setState` here?

elbertcastaneda replied:
I wrote that code because i seen the next TODO : 
```js
    // TODO: Update the state.date only if the date.temperature is different
    // Use functional state

    socket.on("FromAPI", (data) => {
      this.setState(() => null);
    });
```
But if I think about it better, I think it was a typing error ("**date**" really is "**data**").

I'm going to change it, am sorry

elbertcastaneda committed:
I remove "error, info" parameters

I was not use the "error, info" parameters, i remove these parameters from the code

Commit:
I implemented Portal component in the presentational/ModalPortal.js file, and remove 'date' setState

Commit:
Random temperature vales

Change the server api to change temperature values in randow.

Robfz commented on session1/challenge/client/src/js/components/presentational/ModalPortal.js:
```diff
@@ -1,4 +1,5 @@
 import React, { Component } from "react";
+import ReactDOM from 'react-dom';
```
You can use the `createPortal` that is being imported in the next line.

elbertcastaneda replied:
👍

elbertcastaneda committed:
I removed ReactDOM import sentence

Robfz commented:
Great job 💯

bit-dragon commented on session1/challenge/client/src/js/components/container/App.js:
```diff
@@ -133,12 +142,16 @@ class App extends Component {
-
-    socket.on("FromAPI", (data) => {
-      this.setState(() => null);
+        return null;
```
👍

bit-dragon commented on session1/challenge/client/src/js/components/presentational/ModalPortal.js:
```diff
@@ -13,7 +13,16 @@ const Modal = ({ open, onClick, children }) => {
+      <Modal open={open} onClick={onClick}>
+        {children}
+      </Modal>,
+      document.getElementsByTagName('body')[0]
```
We can also use `document.body`.

elbertcastaneda replied:
yes, you're right, the "document.body" is better

bit-dragon commented on session1/challenge/server/app.js:
```diff
@@ -52,6 +52,8 @@ const getApiAndEmit = async socket => {
     // const res = await axios.get(`${API_HOST}${SECRET_KEY}/${LOCATION}`);
     // socket.emit("FromAPI", { data: res.data.currently });
+    const randIndex = Math.floor(Math.random() * parseInt(temperatureValues.length - 1));
+    data.currently.temperature = temperatureValues[randIndex];
```
👍

elbertcastaneda committed:
Suggested change 👍
